### ISSUE TITLE
Add traverse and sequence aliases

### ISSUE THREAD

**Issue by iravid** [zio/zio#1121]
Try as I might, I keep reaching for these names instead of `foreach` and `collect`.

What do everyone think of re-adding these aliases?

A data point in favour of this is the equivalence with `Future.traverse`, `Future.sequence`.

These need to be added for:
- Exit
- ZIO
- ZManaged

**Comment by kaishh** [comment]
I'm all for, obviously. :>

**Comment by ghostdogpr** [comment]
I agree, the question keeps coming back on Gitter, might be the most frequently asked one.

**Comment by hepin1989** [comment]
@iravid So do I, I do not like the current foreach and foreachPar name. for each for me always return unit/void.

**Comment by dsebban** [comment]
Hey would love to take this

**Comment by dsebban** [comment]
Do you think `foreach` and `collectAll` should be deprecated in favor of `traverse` and `sequence` (i.e renaming it all over) ?

**Comment by kaishh** [comment]
I guess ne

**Comment by iravid** [comment]
Let’s keep them for now and let @jdegoes decide on their fate :-)

**Comment by jdegoes** [comment]
The only people who ask about these names are the people from the (existing) pure FP community, who are trained on sequence/traverse. 

I don't mind aliases if others like them, but it's worth keeping in mind:

 - Many people from "pure FP Scala" will be using ZIO with Cats / Scalaz, which already provide (as syntax) these methods, with the names functional programmers are familiar with
 - Most new ZIO users don't use Scalaz or Cats and are not from the pure FP community
 - The pure FP Scala community is < 0.5% of the total market

I beta tested many possible names for these during subsequent _Functional Scala_ courses, and while not perfect, `foreach` / `collectAll` played out better than alternatives. To someone with no background in FP, `traverse` and `sequence` are not meaningful names, whereas, e.g.,`foreach` is a slight generalization on Scala's own `foreach` that collects each expression returned in the body of the for loop; it has significantly better teachability in my classes.

In summary:

 - Good to merge the "classic" aliases if people like them
 - Would not want to remove the old ones or favor the new ones in documentation

**Comment by ghostdogpr** [comment]
I used to know `traverse` and `sequence` long before I did any FP, just because they're named like that on `Future`. So I think the amount of people reaching for it is larger than the FP community.

**Comment by jdegoes** [comment]
@ghostdogpr Good data point! At my workshops, I do polls on this type of stuff. In a given audience, about 30% of people use Future (sometimes 50%). Only 1 - 3 people in any class know how to deal with `Future[Option[A]]` (i.e. know a method that can sequence the result). Of the few people who run into this type, most seem to deal with the problem by hand or cheat.

I have no objection to adding them and I think having aliases that match Future here is probably useful.

**Comment by hepin1989** [comment]
![image](https://user-images.githubusercontent.com/501740/60533722-e22c7200-9d32-11e9-8ab7-811c709295cf.png)

**Comment by ivladpatryshev** [comment]
I wonder if there is a solution for collecting errors (the way `traverse` _was supposed_ to work.

**Comment by adamgfraser** [comment]
@ivladpatryshev You can use `ZIO.validate` if you want to accumulate errors.
